test/splitter.js

```diff
@@ -0,0 +1,160 @@
+
+        assert.strictEqual(aliceEthBalance.toString(10), expectedSenderEthBalance.toString(10));
+
+        // Calculate the expected contract balances
+        const expectedSenderContractBalance = initSenderContractBalance.add(remainder);
+        const expectedRecipientOneContractBalance = initRecipientOneContractBalance.add(splitAmount);
+        const expectedRecipientTwoContractBalance = initRecipientTwoContractBalance.add(splitAmount);
+
+        // Get the actual contract balances
+        const aliceContractBalance = await splitter.balances(senderAddress);
+        const bobContractBalance = await splitter.balances(RecipientOneAddress);
+        const carolContractBalance = await splitter.balances(RecipientTwoAddress);
+
+        assert.strictEqual(aliceContractBalance.toString(10), expectedSenderContractBalance.toString(10));
+        assert.strictEqual(bobContractBalance.toString(10), expectedRecipientOneContractBalance.toString(10));
+        assert.strictEqual(carolContractBalance.toString(10), expectedRecipientTwoContractBalance.toString(10));
+
+        truffleAssert.eventEmitted(receipt, 'Deposit', (ev) => {
+            return  ev.sender === senderAddress && ev.recipient1 === RecipientOneAddress &&
+                    ev.recipient2 === RecipientTwoAddress && ev.amount.toString(10) === "2" &&
+                    ev.remainder.toString(10) === "1";
+        }, 'TestEvent should be emitted with correct parameters');
+    })
+
+    it("Transaction reverts if the deposit amount is zero", async () => {
+        await truffleAssert.reverts(
+            splitter.splitDeposit(RecipientTwoAddress, RecipientOneAddress,{from: senderAddress, value: 0}),
+            "The value must be greater than 0"
+        );
+        checkEventNotEmitted("Deposit");
+    });
+
+    it("Transaction reverts if the first recipient is the same as the second recipient", async () => {
+        await truffleAssert.reverts(
+            splitter.splitDeposit(RecipientOneAddress, RecipientOneAddress,{from: senderAddress, value: 4}),
+            "The first recipient is the same as the second recipient"
+        );
+        checkEventNotEmitted("Deposit");
+    });
+
+    it("Transaction reverts if the sender's address is also a recipient", async () => {
+        await truffleAssert.reverts(
+            splitter.splitDeposit(senderAddress, RecipientTwoAddress, {from: senderAddress, value: 4}),
+            "The sender cannot also be a recipient"
+        );
+        checkEventNotEmitted("Deposit");
+    });
+
+    it('Second recipient can successfully withdraw 2 wei', async () => {
+
+        const depositAmount = web3.utils.toWei(web3.utils.toBN(1), "ether");
+        const withDrawAmount = web3.utils.toBN(2);
+
+        // Take snapshot of the recipients ETH balance
+        let initRecipientTwoEthBalance = await web3.eth.getBalance(RecipientTwoAddress);
+        initRecipientTwoEthBalance = web3.utils.toBN(initRecipientTwoEthBalance);
```

Comment:
To be really sure, I would fetch the recipient's balance right before the withdraw transaction.